guides/metrics.md

```diff
@@ -0,0 +1,75 @@
+    var hostname = env.getOrDefault("HOST", "localhost");
+    var port = Integer.parseInt(env.getOrDefault("PORT", "9200"));
+    var user = env.getOrDefault("USERNAME", "admin");
+    var pass = env.getOrDefault("PASSWORD", "admin");
+    var metricEnabled = true;
+    double PERCENTILE_99 = 0.99;
+    double PERCENTILE_95 = 0.95;
+    var meterRegistry =  new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
+
+    final var hosts = new HttpHost[]{new HttpHost(https ? "https" : "http", hostname, port)};
+
+    final var sslContext = SSLContextBuilder.create().loadTrustMaterial(null, (chains, authType) -> true).build();
+
+    MetricOptions metricOptions = MetricOptions.builder()
+            .setMetricsEnabled(metricEnabled) // required to turn metrics on/off
+            .setMeterRegistry(meterRegistry) 
+            .setPercentiles(PERCENTILE_99, PERCENTILE_95)
+            .setAdditionalMetricGroups(MetricGroup.NETWORK_DETAILS)
+            .setExcludedTags(MetricTag.HOST_CONTACTED)
+            .build();
+
+    final var transport = ApacheHttpClient5TransportBuilder.builder(hosts)
+            .setMapper(new JacksonJsonpMapper())
+            .setMetricOptions(metricOptions)
+            .setHttpClientConfigCallback(httpClientBuilder -> {
+              final var credentialsProvider = new BasicCredentialsProvider();
+              for (final var host : hosts) {
+                credentialsProvider.setCredentials(new AuthScope(host), new UsernamePasswordCredentials(user, pass.toCharArray()));
+              }
+
+// Disable SSL/TLS verification as our local testing clusters use self-signed certificates
+              final var tlsStrategy = ClientTlsStrategyBuilder.create()
+                      .setSslContext(sslContext)
+                      .setHostnameVerifier(NoopHostnameVerifier.INSTANCE)
+                      .build();
+
+              final var connectionManager = PoolingAsyncClientConnectionManagerBuilder.create().setTlsStrategy(tlsStrategy).build();
+
+              return httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider).setConnectionManager(connectionManager);
+            })
+            .build();
+    return new OpenSearchClient(transport);
+  }
+}
+```
+
+## Metrics Collected
+
+The OpenSearch Java client collects a variety of metrics to provide insights into its operations. Below is a summary of the key metrics collected, along with their descriptions and important dimensions, such as status code and request type, that can be used for filtering and analysis.
+
+| Metric                                | Description                                                     | Important Dimensions(tags)     |
+|---------------------------------------|-----------------------------------------------------------------|--------------------------------|
+| os.client.request.seconds             | End-to-end request execution latency                            | StatusCodeOrException, Request |
```

Comment:
nit (also for the other rows): i'd mark the metric name & tags as code for better readability):
```suggestion
| `os.client.request.seconds`             | End-to-end request execution latency                            | `StatusCodeOrException`, `Request` |
```